### ISSUE TITLE
[feature-request] Private Registry

### ISSUE THREAD

**Issue by AntonovRoman** [testcontainers/testcontainers-node#62]
### Overview
It would be nice to have the ability to pull images from private registries the same way as `java` lib does.
https://github.com/testcontainers/testcontainers-java/blob/master/core/src/main/java/org/testcontainers/utility/RegistryAuthLocator.java

- Auth from local `~/.docker/config.json`
- Support for `credHelpers` and `credsStore`

**Comment by cristianrgreco** [comment]
Hi @AntonovRoman 

What do you think about something like:

```javascript
new GenericContainer("private-image")
  .withAuthentication({
    username: '...',
    password: '...',
    email: '...',
    serverAddress: '...'
  })
```

**Comment by AntonovRoman** [comment]
@cristianrgreco That's what i was thinking for a short term solution. I can create a PR for that.

**Comment by cristianrgreco** [comment]
@AntonovRoman Available in 2.8.0

**Comment by AntonovRoman** [comment]
@cristianrgreco thank you !
